components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/input/PasswordInputView.kt

```diff
@@ -0,0 +1,376 @@
+    hintContentDescription: String? = null,
+    prefix: @Composable() (() -> Unit)? = null,
+    placeholderText: String? = null,
+    errorText: String? = null,
+    errorContentDescription: String? = null,
+    showButtonText: String? = stringResource(R.string.passwordTextInputView_show),
+    hideButtonText: String? = stringResource(R.string.passwordTextInputView_hide),
+    showButtonContentDescription: String? = null,
+    hideButtonContentDescription: String? = null,
+    showButtonStateDescription: String? = null,
+    hideButtonStateDescription: String? = null,
+    characterCount: Int? = null,
+    maxChars: Int? = null,
+    singleLine: Boolean = false,
+    keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
+    interactionSource: MutableInteractionSource,
+    requiredSequencesSpacing: Boolean,
+    textStyle: TextStyle
+) {
+    var hideButtonShown: Boolean by rememberSaveable { mutableStateOf(false) }
+
+    var localValue: String by rememberSaveable { mutableStateOf(value.orEmpty()) }
+    localValue = value.orEmpty()
+
+    var localError: String? by rememberSaveable { mutableStateOf(null) }
+    localError = errorText
+
+    var isError: Boolean by rememberSaveable { mutableStateOf(false) }
+
+    isError = !localError.isNullOrEmpty() || (localValue.length > (characterCount ?: Int.MAX_VALUE))
+
+    @Composable
+    fun ShowHideButton(
+        modifier: Modifier,
+        hideButtonContentDescription: String?,
+        hideButtonStateDescription: String?,
+        showButtonContentDescription: String?,
+        showButtonStateDescription: String?,
+        showButtonText: String,
+        hideButtonText: String
+    ) {
+        if (hideButtonShown) {
+            SecondaryButton(
+                hideButtonText, onClick = { hideButtonShown = false },
+                modifier = modifier
+                    .semantics {
+                        hideButtonContentDescription?.let { contentDescription = it }
+                        hideButtonStateDescription?.let { stateDescription = it }
+                    }
+            )
+        } else {
+            SecondaryButton(
+                showButtonText, onClick = { hideButtonShown = true },
+                modifier = modifier
+                    .semantics {
+                        showButtonContentDescription?.let { contentDescription = it }
+                        showButtonStateDescription?.let { stateDescription = it }
+                    }
+            )
+        }
+    }
+
+    CalculateBorder(
+        true,
+        isError,
+        interactionSource,
+        focusedBorderThickness = FocusedBorderThickness,
+        unfocusedBorderThickness = UnfocusedBorderThickness
+    )
+    Column(modifier = modifier) {
+        Label(labelText = labelText, labelContentDescription = labelContentDescription)
+        Hint(hintText = hintText, hintContentDescription = hintContentDescription)
+        Row(
+            borderStroke?.let {
```

Comment:
Should this be nullable? Won't we always have a value from `animateBorderStrokeAsState` which doesn't have a null state. I think we might be able to simplify this.